# Individual Planning Report - Chris Gibbons (Group 2)

In [6]:
library(tidyverse)

players_data <- read_csv("https://raw.githubusercontent.com/chrispy-tacos/dsci_idv_planning/refs/heads/main/players.csv")
sessions_data <- read_csv("https://raw.githubusercontent.com/chrispy-tacos/dsci_idv_planning/refs/heads/main/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


## Data Description

In [7]:
summary(players_data)
unique(players_data$experience)
unique(players_data$gender)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

[1] "Pro"      "Veteran"  "Amateur"  "Regular"  "Beginner"

[1] "Male"              "Female"            "Non-binary"       
[4] "Prefer not to say" "Agender"           "Two-Spirited"     
[7] "Other"

The players data consist of 196 observations, with each corresponding to a unique participant in the PLAICraft server. 7 variables are recorded for each participant:

- **experience:** A qualitative character variable classifying the participant's Minecraft experience as one of 5 categories: 'Beginner', 'Regular', 'Amateur', 'Veteran', or 'Pro'.
  
- **subscribe:** A qualitative logical variable indicating whether or not the participant subscribes to a game-related newspaper.

- **hashedEmail:** A qualitative character variable indicating the participant's encrypted email address.

- **played_hours:** A quantitative double variable measuring the hours the participant has spent playing on the PLAICraft server.

- **name:** A qualitative character variable indicating the participant's name.

- **gender:** A qualitative character variable classifying the participant's gender as one of 7 categories: 'Male', 'Female', 'Non-binary', 'Prefer not to say', 'Agender', 'Two-Spirited', or 'Other'.

- **Age:** A quantitative double variable indicating the participant's age (in years).

Further inspection of this data reveals that the "played_hours" variable contains major outliers. Furthermore, the "experience" and "gender" variables are categorical, and should thus be listed as factor variables.

In [8]:
summary(sessions_data)

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          

The sessions data consist of 1535 observations, with each corresponding to an individual play session of a given participant on the PLAICraft server. 5 variables are recorded for each session:

- **hashedEmail:** A qualitative character variable indicating the encrypted email address of the participant whose session it is.
  
- **start_time:** A qualitative character variable describing the date and time the session started.

- **end_time:** A qualitative character variable describing the date and time the session ended.

- **original_start_time:** A quantitative double variable measuring the exact start of the session in UNIX time.

- **original_end_time:** A quantitative double variable measuring the exact end of the session in UNIX time.

Further inspection of this data reveals that the "start_time" and "end_time" variables record two separate values in each cell: date and time. Therefore, the data is untidy.

## GitHub Repository

Link: https://github.com/chrispy-tacos/dsci_idv_planning.git